lib/DisposedException.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace Amp;
+
+/**
+ * Will be thrown from {@see StreamSource::yield()} or the yield callable provided by {@see AsyncGenerator} if the
+ * associated stream is destroyed.
+ */
+final class DisposedException extends \Exception
```

Comment:
`AlreadyDestroyedException` sounds more self-explaining

Reply:
It's not realy destroyed if we offer a `dispose()` method.